# **CASE STUDY 1**
> *(In Bahasa) for English version
https://github.com/ardisoegi/ai-audit-case-studies/blob/main/Case_Study_1_ENG.ipynb*

> klik **> Run All** di toolbar di atas untuk melihat semua hasil proses perintah sekaligus


# 1. Overview


##**1.1 - Konteks Dataset**

Nama Dataset : **log transaksi karyawan**

Dataset ini merupakan data simulasi untuk keperluan latihan audit yang berasal dari bantuan AI generatif. Audit  ini dilakukan untuk mencari mencari anomali yang disengaja dari Dataset

> *Preparation : Untuk memulai, jalankan kode di bawah ini (shortcut : ctrl + Enter)*













In [ ]:
import pandas as pd
import numpy as np

# Berikut adalah data log transaksi karyawan
data = {
    "employee_id": ["E001", "E002", "E003", "E004", "E005",
                    "E003", "E006", "E007", "E008", "E009"],
    "name": ["Budi Santoso", "Siti Rahma", "Andi Wijaya", "Dewi Kusuma",
              "Reza Pratama", "Andi Wijaya", "Maya Indah",
              "Fajar Nugroho", "Linda Sari", "Hendra Gunawan"],
    "department": ["Finance", "HR", "Finance", "IT", "Finance",
                   "Finance", "Marketing", "", "HR", "IT"],
    "amount": [1500000, 750000, 15000000, None, 1200000,
               9800000, 500000, 650000, -200000, 900000],
    "date": ["2024-01-05", "2024-01-06", "2024-01-07", "2024-01-08",
             "2024-01-09", "2024-01-10", "2024-01-11", "2024-01-12",
             "2024-01-13", "2024-01-14"],
    "status": ["approved", "approved", "approved", "pending",
                "approved", "approved", "approved", None,
                "approved", "rejected"]
}

df = pd.DataFrame(data)

# Output
print(df)

  employee_id            name department      amount        date    status
0        E001    Budi Santoso    Finance   1500000.0  2024-01-05  approved
1        E002      Siti Rahma         HR    750000.0  2024-01-06  approved
2        E003     Andi Wijaya    Finance  15000000.0  2024-01-07  approved
3        E004     Dewi Kusuma         IT         NaN  2024-01-08   pending
4        E005    Reza Pratama    Finance   1200000.0  2024-01-09  approved
5        E003     Andi Wijaya    Finance   9800000.0  2024-01-10  approved
6        E006      Maya Indah  Marketing    500000.0  2024-01-11  approved
7        E007   Fajar Nugroho               650000.0  2024-01-12      None
8        E008      Linda Sari         HR   -200000.0  2024-01-13  approved
9        E009  Hendra Gunawan         IT    900000.0  2024-01-14  rejected


##**1.2 - Struktur Dataset**

In [ ]:
# Metode untuk cek struktur dataset
print(f"Berikut adalah tampilan header dataset")
print('='*50)
print(df.head(1))
print("\n")

print(f"Berikut adalah shape dataset (jumlah baris, jumlah kolom)")
print('='*50)
print(df.shape)
print("\n")

print(f"Berikut adalah tipe dari kolom dataset")
print('='*50)
print(df.dtypes)


Berikut adalah tampilan header dataset
  employee_id          name department     amount        date    status
0        E001  Budi Santoso    Finance  1500000.0  2024-01-05  approved


Berikut adalah shape dataset (jumlah baris, jumlah kolom)
(10, 6)


Berikut adalah tipe dari kolom dataset
employee_id     object
name            object
department      object
amount         float64
date            object
status          object
dtype: object


Dari perintah di atas dihasilkan Struktur Dataset
-  **6** Header; **employee_id, name, department, amount, date, status**
-  **10** Baris, **6** Kolom
-  Tipe data object; **employee_id, name, department, date, status**
-  Tipe data float64 (desimal); **amount**
-  Tipe dataset; **object**

> Dataset sudah teridentifikasi dan bisa dilaksanakan audit.

##**1.3 - Ringkasan Statistik**

In [ ]:
# Metode untuk melihat statistik dari dataset
df.describe()

,amount
count,9.000000e+00
mean,3.344444e+06
std,5.316804e+06
min,-2.000000e+05
25%,6.500000e+05
50%,9.000000e+05
75%,1.500000e+06
max,1.500000e+07


Berikut temuan awal dari statistik dataset


-   Nilai minimum (min) bernilai negatif **-200.000**; ada indikasi anomali
-   Outlier (dari max) **15.000.000**; terlalu jauh rentangnya dari rata-rata (mean) **3.344.444**
-   Jumlah '**amount**' hanya **9**, selisih **1** dari total data yaitu **10**

# 2. Analisis




##**2.1 - Checklist Audit**
1.   Transaksi mana yang jumlahnya di atas rata-rata keseluruhan?
2.   Ada berapa baris yang memiliki nilai kosong (NaN) di kolom mana saja?
3.   Employee mana yang muncul lebih dari satu kali dan apakah total transaksinya wajar?

##**2.2 - Solusi Checklist 1**

Transaksi mana yang jumlahnya di atas rata-rata keseluruhan?



**Checklist 1 - Step 1** : Mencari nilai rata-rata

In [ ]:
# Metode untuk mencari rata-rata (mean)
df_mean_transaksi = df['amount'].mean()

# Output
print(f"Nilai transaksi rata-rata : {df_mean_transaksi}")

**Checklist 1 - Step 2** : Mencari data yang nilai 'amount' melebihi nilai rata-rata


In [ ]:
# Metode mencari nilai amount > dari mean
df_atas_rata = df[df['amount'] > df_mean_transaksi]

# Output
print(df_atas_rata)


###**Checklist 1 - Conclusion**

Ditemukan **2 transaksi** yang melebihi nilai rata-rata pada :

1. index 2;        E003  Andi Wijaya    Finance  15000000.0  2024-01-07  approved
2. index 5;        E003  Andi Wijaya    Finance   9800000.0  2024-01-10  approved


---




##**2.3 - Solusi Checklist 2**

Ada berapa baris yang memiliki nilai kosong (NaN) di kolom mana saja?


**Checklist 2 - Step 1** : Mencari nilai kosong dataset

In [ ]:
# Metode mengubah semua string kosong menjadi NaN
df['department'] = df['department'].replace("", np.nan)

# Metode mencari nilai kosong
df_nilai_kosong = df.isnull().sum()

# Output
print(df_nilai_kosong)

employee_id    0
name           0
department     1
amount         1
date           0
status         1
dtype: int64


###**Checklist 2 - Conclusion**

Jadi, ditemukan nilai kosong di kolom **amount** sebanyak **1**, kolom **department** sebanyak **1**, dan  kolom **status** sebanyak **1**


---



##**2.4 - Solusi Checklist 3**

Employee mana yang muncul lebih dari satu kali — dan apakah total transaksinya wajar?


**Checklist 3 - Step 1** : Mencari nama yang sering muncul

In [ ]:
# Metode megelompokkan data nama berdasarkan employee_id
df_karyawan_sering_muncul = df.groupby('employee_id')['name'].value_counts()

# Output
print(df_karyawan_sering_muncul)

**Checklist 3 - Step 2** : Mencari nama karyawan berdasarkan ID yang transaksinya tidak wajar, lalu menilai anomali transaksinya

In [ ]:
# Metode mengelompokkaan per employee_id dan menjumlah nilai amount
df_total_per_karyawan = df.groupby('employee_id').agg({'name': 'first', 'amount': 'sum'})

# Output
print(df_total_per_karyawan)

###**Checklist 3 - Conclusion**

Ditemukan ID **E003** nama **Andi Wijaya** melakukan **2 kali** transaksi


---




#3. Laporan Audit



##**3.1 - Temuan**

1.   Karyawan nama **Andi Wijaya**, melakukan **2 kali** transaksi (**15.000.000** dan **9.800.000**) dengan total **24.800.000** melebihi batas (Outlier) dari rata-rata karyawan lain yaitu **3.344.444** per transaksi. Ada indikasi juga karyawan ini melakukan **Splitting Transaction** yang bertujuan menghindari batasan transaksi (Transaction Threshold). Perlu ditindaklanjuti anomali ini karena langsung berimbas ke nilai transaksi. Resiko : **High**
2.   Karyawan nama **Dewi Kusuma** memiliki data transaksi tidak valid berupa nilai kosong **NaN** (Not a Number). Perlu diselidiki, apakah terdapat kesalahan dalam proses penginputan. Resiko : **High**
3.   Karyawan nama **Linda Sari** memiliki data transaksi **Negatif** yaitu **-200.000**. Apakah ini legitimate/valid? Resiko : **Medium**
4.   karyawan nama **Fajar Nugroho** tidak memiliki value di kolom **departemen**. Dan juga memiliki status **none**. Beresiko diabaikan sistem.
Resiko : **Medium**



##**3.2 - Rekomendasi**

1.   Investigasi dan verifikasi transaksi anomali pada karyawan bernama Andi Wijaya. Proses ini bisa ditindaklanjuti oleh departemen Finance/HR.
2.   Tentukan aturan untuk tipe input di kolom yang seharusnya dalam type nominal angka. Ditujukan untuk departemen Developer/Engineer untuk membuat perbaikan.
3.   Kebijakan nilai minus perlu ditentukan ke-valid-annya. Regulasi diperlukan oleh Finance/HR
4.   Tentukan standar kolom kosong pada nilai status yang di luar kategori. Untuk nilai di luar kategori seharusnya diseragamkan misalnya dari tanpa nilai diganti menjadi "None". Developer/Engineer perlu melakukan adjustment.


##**3.3 - Limitasi**

Dataset ini hanya berisi 10 baris data dan merupakan hasil dari bantuan AI generatif dengan tujuan menjadi bahan latihan audit; temuan bias transaksi perlu diverifikasi dengan data yang lebih besar sebelum bisa dijadikan kesimpulan.